MetroidPrimeClient.py

```diff
@@ -0,0 +1,206 @@
+    energy_tank_item = current_items["Energy Tank"]
+    num_energy_tanks_received = get_total_count_of_item_received(
+        energy_tank_item.code, ctx.items_received)
+    diff = num_energy_tanks_received - energy_tank_item.current_capacity
+    if diff > 0 and energy_tank_item.current_capacity < energy_tank_item.max_capacity:
+        new_capacity = min(num_energy_tanks_received,
+                           energy_tank_item.max_capacity)
+        logger.debug(
+            f"Setting energy tanks to {new_capacity} from {energy_tank_item.current_capacity}")
+        ctx.game_interface.give_item_to_player(
+            energy_tank_item.id, new_capacity, new_capacity)
+
+        # Heal player when they receive a new energy tank
+        # Player starts with 99 health and each energy tank adds 100 additional
+        ctx.game_interface.set_current_health(new_capacity * 100.0 + 99)
+
+    # TODO: Handle setting Artifact flags so that the Artifact Temple State is updated accordingly
+
+
+async def _handle_game_ready(ctx: MetroidPrimeContext):
+    if ctx.server:
+        if not ctx.slot:
+            await asyncio.sleep(1)
+            return
+
+        await send_checked_locations(ctx)
+        await handle_receive_items(ctx)
+
+        if "DeathLink" in ctx.tags:
+            logger.debug("DeathLink not implemented")
+        await asyncio.sleep(0.5)
+    else:
+        logger.info("Waiting for player to connect to server")
+        await asyncio.sleep(1)
+
+
+async def _handle_game_not_ready(ctx: MetroidPrimeContext):
+    """If the game is not connected or not in a playable state, this will attempt to retry connecting to the game."""
+    if not ctx.game_interface.is_connected():
+        logger.info("Attempting to connect to Dolphin")
+        ctx.game_interface.connect_to_game()
+    elif not ctx.game_interface.is_in_playable_state():
+        logger.info(
+            "Waiting for player to load a save file or start a new game")
+        await asyncio.sleep(3)
+
+
+def main(connect=None, password=None, name=None):
+    Utils.init_logging("Metroid Prime Client")
+
+    async def _main(connect, password, name):
+        ctx = MetroidPrimeContext(connect, password)
+        ctx.auth = name
+        ctx.server_task = asyncio.create_task(
+            server_loop(ctx), name="ServerLoop")
+        if gui_enabled:
+            ctx.run_gui()
+        await asyncio.sleep(1)
+
+        ctx.dolphin_sync_task = asyncio.create_task(
+            dolphin_sync_task(ctx), name="DolphinSync")
+
+        await ctx.exit_event.wait()
+        ctx.server_address = None
+
+        await ctx.shutdown()
+
+        if ctx.dolphin_sync_task:
+            await asyncio.sleep(3)
+            await ctx.dolphin_sync_task
+
+    import colorama
+
+    colorama.init()
+    asyncio.run(_main(connect, password, name))
+    colorama.deinit()
+
+
+if __name__ == "__main__":
+    parser = get_base_parser()
+    parser.add_argument('--name', default=None,

```

Comment:
Added support for a `--name` argument so I don't have to type it in every time I test the client connecting to AP. SC2 client does something similar so I figured it wasn't going to break anything upstream 🤣